components/index/LocationInfo.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import { Popover } from "flowbite-react";
+import { FiInfo } from "react-icons/fi";
+
+type LocationInfoProps = {
+  locationName: string;
+  locationDetails: {
+    display_name: string | null;
+  };
+};
+
+const LocationInfo: React.FC<LocationInfoProps> = ({
+  locationName,
+  locationDetails,
+}) => {
+  const parseLocation = (display_name: string | null) => {
```

Comment:
I think this parseLocation logic is too weak. The string is not always guaranteed to be formatted this way